src/functions.js

```diff
@@ -214,36 +214,90 @@ var _ = Mavo.Functions = {
 	 * Aggregate sum
 	 */
 	sum: function(array) {
-		return $u.numbers(array, arguments).reduce((prev, current) => {
-			return +prev + (+current || 0);
-		}, 0);
+		if (array[0] && array[0]["$groupedBy"] === Mavo.groupedBy) { // grouped structures
+			var ret = [];
+
+			for (i in array) {
+				ret.push(_.sum(array[i]["$items"]));
+			}
+
+			return ret;
+		}
+		else {
+			return $u.numbers(array, arguments).reduce((prev, current) => {
+				return +prev + (+current || 0);
+			}, 0);
+		}
 	},
 
 	/**
 	 * Average of an array of numbers
 	 */
 	average: function(array) {
 		array = $u.numbers(array, arguments);
+		if (array[0] && array[0]["$groupedBy"] === Mavo.groupedBy) { // grouped structures
+			var ret = [];
+
+			for (i in array) {
+				ret.push(_.average(array[i]["$items"]));
+			}
 
-		return array.length && _.sum(array) / array.length;
+			return ret;
+		}
+		else {
+			return array.length && _.sum(array) / array.length;
+		}
 	},
 
 	/**
 	 * Min of an array of numbers
 	 */
 	min: function(array) {
-		return Math.min(...$u.numbers(array, arguments));
+		if (array[0] &&array[0]["$groupedBy"] === Mavo.groupedBy) { // grouped structures
+			var ret = [];
+
+			for (i in array) {
+				ret.push(_.min(array[i]["$items"]));
+			}
+
+			return ret;
+		}
+		else {
+			return Math.min(...$u.numbers(array, arguments));
+		}
 	},
 
 	/**
 	 * Max of an array of numbers
 	 */
 	max: function(array) {
-		return Math.max(...$u.numbers(array, arguments));
+		if (array[0] &&array[0]["$groupedBy"] === Mavo.groupedBy) { // grouped structures
+			var ret = [];
+
+			for (i in array) {
+				ret.push(_.max(array[i]["$items"]));
+			}
+
+			return ret;
+		}
+		else {
+			return Math.max(...$u.numbers(array, arguments));
+		}
 	},
 
 	count: function(array) {
-		return Mavo.toArray(array).filter(a => !empty(a)).length;
+		if (array[0] && array[0]["$groupedBy"] === Mavo.groupedBy) { // grouped structures
+			var ret = [];
+
+			for (i in array) {
```

Comment:
Avoid iterating over arrays with a `for..in` loop. They can often have other enumerable properties beyond their indices.
Prefer normal `for` loops, or `forEach()`.